File: sanghoon/programmers-2/프로세스.py
from collections import deque
def solution(priorities, location):
    queue = []
    for process in enumerate(priorities) :     # index, 값 형태로 만들기
        queue.append(process)

    answer = 0
    while True:
        current = queue.pop(0)                      # 현재 프로세스에 올라간 인덱스, 값
        if any(current[1] < q[1] for q in queue):   # 현재 프로세스 우선순위보다 높은 우선 순위 찾기
            queue.append(current)                   # 제일 뒤 후순위로 보냄
        else:
            answer += 1                             # 카운트 시작
            if current[0] == location:              # index와 동일한 location 일 경우
                return answer

priorities = [1, 1, 9, 1, 1, 1]
location = 0
solution(priorities, location)

File: kyulee/programmers-2/프로세스.java

import java.util.LinkedList;
import java.util.Queue;

public class 프로세스 {
    class Solution {
        public int solution(int[] priorities, int location) {
            Queue<Integer> queue = new LinkedList<>();

            for (int priority : priorities) {
                queue.offer(priority);
            }

            int answer = 0;

            while (!queue.isEmpty()) {
                int current = queue.poll();
                boolean isPrinted = true;

                for (int element : queue) {
                    if (current < element) {
                        isPrinted = false;
                        break;
                    }
                }

                if (isPrinted) {
                    answer++;

                    if (location == 0) {
                        break;
                    } else {
                        location--;
                    }
                } else {
                    queue.offer(current);

                    if (location == 0) {
                        location = queue.size() - 1;
                    } else {
                        location--;
                    }
                }
            }

            return answer;
        }
    }

    public class Main {
        public static void main(String[] args) {
            Solution solution = new Solution();

            // 예시 테스트
            int[] priorities1 = {2, 1, 3, 2};
            int location1 = 2;
            System.out.println(solution.solution(priorities1, location1));  // 1

            int[] priorities2 = {1, 1, 9, 1, 1, 1};
            int location2 = 0;
            System.out.println(solution.solution(priorities2, location2));  // 5
        }
    }
}
